ui/src/components/pipeline/edgeinfo/EdgeInfo.tsx

```diff
@@ -123,39 +146,47 @@ export default function EdgeInfo(props: EdgeInfoProps) {
         </TabPanel>
       )}
 
-      {edge?.data?.edgeWatermark &&
-          <TabPanel value={value} index={1}>
-            <TableContainer
-                component={Paper}
-                sx={{ borderBottom: 1, borderColor: "divider", width: 400 }}
-            >
-              <Table aria-label="buffer-watermark">
-                <TableHead>
-                  <TableRow>
-                    <TableCell>Partition</TableCell>
-                    <TableCell >Watermark</TableCell>
-                  </TableRow>
-                </TableHead>
-                <TableBody>
-                  {edge?.data?.edgeWatermark?.watermarks &&
-                      edge.data.edgeWatermark.watermarks.map((Watermark, idx) => (
-                          <TableRow>
-                            <TableCell >{idx}</TableCell>
-                            <TableCell >{Watermark} ({new Date(Watermark).toISOString()})</TableCell>
-                          </TableRow>
-                      ))}
-                </TableBody>
-              </Table>
-            </TableContainer>
-          </TabPanel>
-      }
+      {edge?.data?.edgeWatermark && (
+        <TabPanel value={value} index={1}>
+          <TableContainer
+            component={Paper}
+            sx={{ borderBottom: 1, borderColor: "divider", width: 400 }}
+          >
+            <Table aria-label="buffer-watermark">
+              <TableHead>
+                <TableRow>
+                  <TableCell>Partition</TableCell>
+                  <TableCell>Watermark</TableCell>
+                </TableRow>
+              </TableHead>
+              <TableBody>
+                {edge.data.edgeWatermark?.watermarks &&
+                  edge.data.edgeWatermark.watermarks.map((Watermark, idx) => (
+                    <TableRow key={idx}>
+                      <TableCell>{idx}</TableCell>
+                      <TableCell>
+                        {Watermark} ({new Date(Watermark).toISOString()})
+                      </TableCell>
+                    </TableRow>
+                  ))}
+              </TableBody>
+            </Table>
+          </TableContainer>
+        </TabPanel>
+      )}
 
       {edge?.data?.conditions && (
         <TabPanel value={value} index={2}>
-          {edges.map((singleEdge) => {
-            if (singleEdge?.data?.conditions && (singleEdge?.source == edge?.data?.fromVertex && singleEdge?.target == edge?.data?.toVertex)) {
-              return <ReactJson
+          {edges.map((singleEdge, idx) => {
+            if (
+              singleEdge?.data?.conditions &&
+              singleEdge?.source == edge?.data?.fromVertex &&
+              singleEdge?.target == edge?.data?.toVertex
+            ) {
+              return (
+                <ReactJson
                   name="conditions"
+                  key={idx}
```

Comment:
idx might be too generic. Can you append specific string also for that component